zrml/prediction-markets/src/lib.rs

```diff
@@ -2345,6 +2346,25 @@ mod pallet {
             Ok(total_weight)
         }
 
+        pub(crate) fn set_market_end(market_id: &MarketIdOf<T>) -> Result<Weight, DispatchError> {
+            <zrml_market_commons::Pallet<T>>::mutate_market(market_id, |market| {
+                market.period = match market.period {
+                    MarketPeriod::Block(ref range) => {
+                        let current_block = <frame_system::Pallet<T>>::block_number();
+                        MarketPeriod::Block(range.start..current_block)
+                    }
+                    MarketPeriod::Timestamp(ref range) => {
+                        let current_time_frame = Self::calculate_time_frame_of_moment(
+                            <zrml_market_commons::Pallet<T>>::now(),
+                        );
+                        MarketPeriod::Timestamp(range.start..current_time_frame)
```

Comment:
The `Timestamp` variant of `MarketPeriod` actually holds the UNIX timestamp (in ms), not the time frame.

```suggestion
                        let now = <zrml_market_commons::Pallet<T>>::now();
                        MarketPeriod::Timestamp(range.start..now)
```